app/src/main/java/space/taran/arknavigator/mvp/presenter/ResourcesPresenter.kt

```diff
@@ -136,14 +142,26 @@ class ResourcesPresenter(
         }
     }
 
-    fun onBackClick() {
+    fun onBackClick() = presenterScope.launch {
         if (!tagsSelectorPresenter.onBackClick())
             router.exit()
     }
 
-    private fun onSelectionChange(selection: Set<ResourceId>) {
+    suspend fun onRemovedResourceDetected() {
+        viewState.setProgressVisibility(true, "Indexing")
+
+        index.reindex()
+        // update current tags storage
+        tagsStorageRepo.provide(rootAndFav)
```

Comment:
Am I right this line triggers garbage collection in tag storage?

Reply:
Yes, also registers new resources. I think we need a clearer way to do this. Perhaps the tags storage should have a reference to the index.